swap_meet/vendor.py

```diff
@@ -1,2 +1,65 @@
 class Vendor:
-    pass
+
+    def __init__(self, inventory = None):
+        if not inventory:
+            inventory = []
+        self.inventory = inventory
+
+    def add(self, new_item):
+        self.new_item = new_item
+        self.inventory.append(new_item)
+        return new_item
+
+    def remove(self, old_item):
+        self.old_item = old_item
+        if old_item in self.inventory:
+            self.inventory.remove(old_item)
+            return old_item
+        return False
+
+    def get_by_category(self, category_title): 
+        self.category_title = category_title
+        clothing_by_category = []
+        for item in self.inventory:
+            if item.category == category_title:
+                clothing_by_category.append(item)
+        return clothing_by_category
+
+    def swap_items(self, friend_vendor, my_item, their_item):
+        if my_item not in self.inventory or their_item not in friend_vendor.inventory:
+            return False
+        friend_vendor.inventory.append(my_item)
+        self.inventory.remove(my_item)
+        self.inventory.append(their_item) 
+        friend_vendor.inventory.remove(their_item)    
```

Comment:
Using the vendor .add and .remove functions here would add another layer of error checking.